### PR TITLE
Updated hillas.py

### PR DESCRIPTION
**Pull request by wrijupan**
- Higher order image parameters added to hillas_parameters_1.
- Fixed inconsistency in length, width in hillas_1 and hillas_2
- Corrected the miss implementation in hillas_1
- For all other changes please see changelog at the top of the script
- Added a preliminary test of hillas.py

### CODE REVIEW

File: ctapipe/reco/hillas.py [hunk 1]
```diff
@@ -71,44 +82,85 @@ def hillas_parameters_1(pix_x, pix_y, image):
...
+    S_x4 = np.sum(image * (pix_x - mean_x) ** 4 ) / size
+    S_y4 = np.sum(image * (pix_y - mean_y) ** 4 ) / size
+
+    #Sanity check2:
+
+    #If S_xy=0 (which should happen not very often, because Size>0) we cannot calculate Length and Width.
+    #In reallity it is almost impossible to have a distribution of cerenkov photons in the used pixels which is exactly symmetric
+    # along one of the axis.
+    if S_xy == 0:
+       print("X and Y uncorrelated. Cannot calculate lenght & width. Exiting ...")
```
Comment (kosack) [inline ctapipe/reco/hillas.py]: this should raise an exception, not print and exit. Ideally,  define a custom exception like `HillasParameterizationError` that is a subclass of `RuntimeError`  and raise it here, with the description as the same text you would have printed.

**Comment by kosack** [comment]
Please merge in the latest changes to hillas.py that were made while you were editing this (I don't think there is a major change, but it can't be merged automatically as it is).  E.g. `git pull origin master` then `git mergetool` (or just a normal `git merge` and then edit the files to remove the conflict markers)

**Comment by wrijupan** [comment]
Thank you for your suggestion Karl. I have now removed the conflict marker and committed the changes to hillas.py.

File: ctapipe/reco/hillas.py [hunk 2]
```diff
@@ -73,44 +87,84 @@ def hillas_parameters_1(pix_x, pix_y, image):
...
-    d = S_yy - S_xx
-    temp = d * d + 4 * S_xy * S_xy
-    a = (d + np.sqrt(temp)) / (2 * S_xy)
-    b = m_y - a * m_x
+    size = np.sum(image)
+
+    #Sanity check1:
+    if size == 0:
+       print ("Empty pixels!")
+       sys.exit(1)
```
Comment (kosack) [inline ctapipe/reco/hillas.py]: please change this to raise a `HillasParameterizationError` as in the other sanity checks

Comment (wrijupan) [inline ctapipe/reco/hillas.py]: Ya thanks. I have made a new commit with the changes.

**Pull request merged**